lib/class-wp-rest-url-details-controller.php

```diff
@@ -343,13 +353,13 @@ private function get_cache( $key ) {
 	/**
 	 * Utility function to cache a given data set at a given cache key.
 	 *
-	 * @param string $key the cache key under which to store the value.
-	 * @param string $data the data to be stored at the given cache key.
-	 * @return void
+	 * @param string $key  The cache key under which to store the value.
+	 * @param string $data The data to be stored at the given cache key.
+	 * @return bool True when transient set, or false.
 	 */
 	private function set_cache( $key, $data = '' ) {
 		if ( ! is_array( $data ) ) {
-			return;
+			return false;
```

Comment:
Looking at this check again, it's not needed and is blocking caching. Why? It's current use case is with string data, as `wp_remote_retrieve_body` returns a string (not an array).

Reply:
Was this lack of caching not exposed by the tests?

Reply:
No not from tests. I found it when doing method signature checks for matching data types. The incoming data is a string. This guard clause bails out if not an array. This means caching is never reaches.

Removed the guard clause. The data is now being cached.

Note: Add a test to ensure being cached and avoid skipping the cache in future iterations.